brainscore/benchmarks/moeller2017.py

```diff
@@ -0,0 +1,503 @@
+            ymin, ymax = np.floor(np.min(y)), np.ceil(np.max(y))
+            grids = np.array(np.meshgrid(np.arange(xmin, xmax, res), np.arange(ymin, ymax, res)))
+            gridx = grids[0].flatten().reshape(-1, 1)
+            gridy = grids[1].flatten().reshape(-1, 1)
+            return gridx, gridy
+
+        # Get voxel coordinates
+        x, y = assembly.neuroid.tissue_x.values, assembly.neuroid.tissue_y.values
+        gridx, gridy = _get_grid_coord()
+
+        # compute sigma from fwmh
+        sigma = fwhm / np.sqrt(8. * np.log(2))
+
+        # define gaussian kernel
+        d_square = (x - gridx.T) ** 2 + (y - gridy.T) ** 2
+        gf = 1. / (2 * np.pi * sigma ** 2) * np.exp(- d_square / (2 * sigma ** 2))
+
+        features_smoothed = DataArray(
+            data=np.dot(assembly.values, gf),
+            dims=['category_name', 'neuroid_id'],
+            coords={'category_name': assembly.category_name.values,
+                    'neuroid_id': assembly.neuroid_id.values,
+                    'voxel_x': ('neuroid_id', gridx.squeeze()),
+                    'voxel_y': ('neuroid_id', gridy.squeeze())}
+        )
+        return features_smoothed,
+
+    @staticmethod
+    def _get_purity_center(selectivity_assembly, radius=1):
+        '''
+        Adapted from Hyo's 2020 paper code
+        Computes the voxel of the selectivity map with the highest purity
+        :param: selectivity_assembly: DataArray
+            dims: 'neuroid_id'
+                    coords:
+                        voxel_x: voxel coordinate
+                        voxel_y: voxel coordinate
+                  'category_name'
+        :param: radius (scalar): radius in mm of the circle in which to consider units
+        :return: (int,int) location of highest purity
+        '''
+
+        def get_purity(center_x, center_y):
+            '''
+            Evaluates purity at a given center position, radius, and corresponding selectivity values
+            '''
+            passing_indices = np.where(np.sqrt(np.square(x - center_x) + np.square(y - center_y)) < radius)[0]
+            return 100. * np.sum(selectivity_assembly.values[passing_indices]) / passing_indices.shape[0]
+
+        x, y = selectivity_assembly.voxel_x.values, selectivity_assembly.voxel_y.values
+        purity = np.array(list(map(get_purity, x, y)))
+        highest_purity_idx = np.argmax(purity)
+
+        center_x, center_y = x[highest_purity_idx], y[highest_purity_idx]
+        return center_x, center_y
+
+    @staticmethod
+    def _determine_face_selectivity(recordings):
+        '''
+        Determines face selectivity of each neuroid
+        :param recordings: DataArray
+        :return: DataArray, same as recordings where activations have been replaced with dprime values
+        '''
+
+        def mean_var(neuron):
+            mean, var = np.mean(neuron.values), np.var(neuron.values)
+            return mean, var
+
+        assert (recordings >= 0).all()

```

Comment:
add error message why <0 is bad, e.g. `assert (recordings >= 0).all(), "I do not like negativity"`